.github/workflows/ci.yml

```diff
@@ -0,0 +1,110 @@
+name: CI
+on: [pull_request, push]
+
+env:
+  ### compiler options
+  HOST:
+  WRAPPER_CMD:
+  # Specific warnings can be disabled with -Wno-error=foo.
+  # -pedantic-errors is not equivalent to -Werror=pedantic and thus not implied by -Werror according to the GCC manual.
+  WERROR_CFLAGS: '-Werror -pedantic-errors'
+  MAKEFLAGS: '-j4'
+  BUILD: 'check'
+  ### secp256k1 config
+  ECMULTWINDOW: 'auto'
+  ECMULTGENPRECISION: 'auto'
+  ASM: 'no'
+  WIDEMUL: 'auto'
+  WITH_VALGRIND: 'yes'
+  EXTRAFLAGS:
+  ### secp256k1 modules
+  EXPERIMENTAL: 'no'
+  ECDH: 'no'
+  RECOVERY: 'no'
+  SCHNORRSIG: 'no'
+  ELLSWIFT: 'no'
+  ### test options
+  SECP256K1_TEST_ITERS:
+  BENCH: 'yes'
+  SECP256K1_BENCH_ITERS: 2
+  CTIMETESTS: 'yes'
+  # Compile and run the examples.
+  EXAMPLES: 'yes'
+
+jobs:
+  modified_files:
+    runs-on: ubuntu-latest
+    outputs:
+      all: ${{ steps.changes.outputs.modified_files}}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Get modified files
+        id: changes
+        env:
+          COMMIT_BEFORE: ${{ github.event_name != 'pull_request' && github.event.pull_request.base.sha || (github.event.before != '0000000000000000000000000000000000000000' && github.event.before || github.event.repository.default_branch) }}
+        run: |
+          git fetch --depth=1 origin "$COMMIT_BEFORE"
+          echo "modified_files=$(git diff --name-only FETCH_HEAD ${{ github.sha }} | xargs)" >> "$GITHUB_OUTPUT"
+
+  docker_image:
+    uses: ./.github/workflows/docker.yml
+    permissions:
+      packages: write
+    needs: modified_files
+    if: contains(needs.modified_files.outputs.all, '.github/workflows/docker.yml') || contains(needs.modified_files.outputs.all, 'ci/linux-debian.Dockerfile')
+
+  x86_64_debian_stable:
+    name: "x86_64: Linux (Debian stable)"
+    runs-on: ubuntu-latest
+    container: ghcr.io/${{ github.repository_owner }}/secp256k1-ci-image:latest
```

Comment:
> how will this work, when a pull modifies the `latest` tag and breaks all other build, because they skip the `docker_image` step?

Do you mean concurrent builds?

Reply:
I see. Going to address this issue shortly.

Reply:
Thanks! [Fixed](https://github.com/bitcoin-core/secp256k1/pull/1396#issuecomment-1668502527).